kedro-datasets/tests/pandas/test_deltatable_dataset.py

```diff
@@ -0,0 +1,147 @@
+from unittest.mock import patch
+
+import pandas as pd
+import pytest
+from deltalake import DataCatalog
+from kedro.io import DataSetError
+from pandas.testing import assert_frame_equal
+
+from kedro_datasets.pandas import DeltaTableDataSet
+
+
+@pytest.fixture
+def filepath(tmp_path):
+    return (tmp_path / "test-delta-table").as_posix()
+
+
+@pytest.fixture
+def dummy_df():
+    return pd.DataFrame({"col1": [1, 2], "col2": [4, 5], "col3": [5, 6]})
+
+
+@pytest.fixture
+def deltatable_data_set_from_path(filepath, load_args, save_args, fs_args):
+    return DeltaTableDataSet(
+        filepath=filepath,
+        load_args=load_args,
+        save_args=save_args,
+        fs_args=fs_args,
+    )
+
+
+class TestDeltaTableDataSet:
```

Comment:
Can you create some test for the object storage? You can refer to `CSVDataSet` using `mock_s3` we can have a local test.

Reply:
It seems that we have difficulty mocking AWS SDK calls from the python side since `deltalake` actually use Rust to Handle the API calls behind the scene. I tried the `moto` way of mocking S3 but it won't work.

Reply:
I believe I can mock this `_internal.RawDeltaTable` for sure, but in that case it won't have a proper local boto3 behavior like the one introduced by `moto` to properly simulate the behavior, unless we also code them all. wdyt?

Reply:
I see, in that case it's fine to leave the unit tests for now. Did you got a chance to test it manually instead?

Reply:
Yes, It works when I test it with my actual AWS account/session. :)